sdk/eventhub/azure-eventhubs/azure/eventhub/aio/eventprocessor/event_processor.py

```diff
@@ -0,0 +1,271 @@
+                task.cancel()
+
+    def _create_tasks_for_claimed_ownership(self, to_claim_ownership_list):
+        for ownership in to_claim_ownership_list:
+            partition_id = ownership["partition_id"]
+            if partition_id not in self._tasks:
+                self._tasks[partition_id] = get_running_loop().create_task(self._receive(ownership))
+
+    async def _receive(self, ownership):
+        log.info("start ownership, %r", ownership)
+        partition_processor = self._partition_processor_factory()
+        partition_id = ownership["partition_id"]
+        eventhub_name = ownership["eventhub_name"]
+        consumer_group_name = ownership["consumer_group_name"]
+        owner_id = ownership["owner_id"]
+        partition_context = PartitionContext(
+            eventhub_name,
+            consumer_group_name,
+            partition_id,
+            owner_id,
+            self._partition_manager
+        )
+        partition_consumer = self._eventhub_client.create_consumer(
+            consumer_group_name,
+            partition_id,
+            EventPosition(ownership.get("offset", self._initial_event_position.value))
+        )
+
+        async def process_error(err):
+            log.warning(
+                "PartitionProcessor of EventProcessor instance %r of eventhub %r partition %r consumer group %r"
+                " has met an error. The exception is %r.",
+                owner_id, eventhub_name, partition_id, consumer_group_name, err
+            )
+            try:
+                await partition_processor.process_error(err, partition_context)
+            except Exception as err_again:  # pylint:disable=broad-except
+                log.warning(
+                    "PartitionProcessor of EventProcessor instance %r of eventhub %r partition %r consumer group %r"
+                    " has another error during running process_error(). The exception is %r.",
+                    owner_id, eventhub_name, partition_id, consumer_group_name, err_again
+                )
+
+        async def close(reason):
+            log.info(
+                "PartitionProcessor of EventProcessor instance %r of eventhub %r partition %r consumer group %r"
+                " is being closed. Reason is: %r",
+                owner_id, eventhub_name, partition_id, consumer_group_name, reason
+            )
+            try:
+                await partition_processor.close(reason, partition_context)
+            except Exception as err:  # pylint:disable=broad-except
+                log.warning(
+                    "PartitionProcessor of EventProcessor instance %r of eventhub %r partition %r consumer group %r"
+                    " has an error during running close(). The exception is %r.",
+                    owner_id, eventhub_name, partition_id, consumer_group_name, err
+                )
+
+        try:
+            while True:
+                try:
+                    await partition_processor.initialize(partition_context)
```

Comment:
should we move this `await partition_processor.initialize(partition_context)` outside of the `while True:`? otherwise in each loop it is called.